src/DataStructures.jl

```diff
@@ -94,4 +94,6 @@ module DataStructures
     export status
     export deref_key, deref_value, deref, advance, regress
 
+    @deprecate iter(s::Stack) collect(s)
+    @deprecate iter(q::Queue) collect(q)
```

Comment:
yea, I was thinking the same thing - not sure which is the best way to go but I suppose just using `s` and `q` is probably best since that's the behavior we'll want to enforce going forward.